src/reducers/org.js

```diff
@@ -893,23 +893,28 @@ export const setSearchFilterInformation = (state, action) => {
   const headers = state.get('headers');
   state = state.asMutable();
 
+  let searchFilterValid = true;
   let searchFilterExpr;
   try {
     searchFilterExpr = headline_filter_parser.parse(searchFilter);
     state.setIn(['search', 'searchFilterExpr'], searchFilterExpr);
-  } catch {
-    // No need to print this parser exceptions.
-    // They are expected, see *.grammar.pegjs
+  } catch (e) {
+    // No need to print this parser exceptions. They are expected, see
+    // *.grammar.pegjs. However, we don't need to update the filtered
+    // headers when given an invalid search filter.
+    searchFilterValid = false;
   }
 
-  state.setIn(['search', 'searchFilterValid'], !!searchFilterExpr);
-
-  let filteredHeaders = headers;
-  if (searchFilterExpr) {
-    // Only run filter if a filter is given and parsing was successfull
-    filteredHeaders = headers.filter(isMatch(searchFilterExpr));
+  state.setIn(['search', 'searchFilterValid'], searchFilterValid);
+  if (searchFilterValid) {
```

Comment:
@schoettl After moving more code out of the try/catch block, we introduced issue #200. I'm adding an additional conditional to remedy. Not sure if more conditionals are better than bigger try/catch blocks, though^^

Reply:
Oh, I considered that case, but apparently had a wrong implementation! I thought `searchFilterExpr` evaluates to false if parse fails, because it should be unassigned in that case. So now there are two `if`s to check for the same thing. And I don't understand why my solution not works. See line 911/912 in this diff.

Reply:
You're right, there's a redundant conditonal. I've remove it.

> And I don't understand why my solution not works. See line 911/912 in this diff.

It didn't work, because in 907 `filteredHeaders` where set to `headers`. They where only conditionally filtered, but in any case they were assigned to the state as `filteredHeaders`. Hence, if the parsing failed, we'd end up with all the headers.

Thanks for checking the diff and hinting to the redundant conditional. I think this is good to merge now^^